src/firefly/java/edu/caltech/ipac/firefly/server/query/AsyncTapQuery.java

```diff
@@ -124,32 +124,65 @@ public Phase getPhase() throws DataAccessException {
         }
 
         public String getErrorMsg()  {
-            try {
-                DataGroup[] results = VoTableReader.voToDataGroups(baseJobUrl + "/error", false);
-                DataGroup errorTbl = results[0];
-                return errorTbl.getAttribute("QUERY_STATUS");
-            } catch (IOException e) {
-                logger.error(e);
-            } catch (DataAccessException e) {
-                return e.getMessage();
+            String errorUrl = baseJobUrl + "/error";
+
+            Ref<String> err = new Ref<>();
+            HttpServices.Status status = HttpServices.getData(errorUrl, HttpServiceInput.createWithCredential(),
+                    (method -> {
+                        boolean decompress = false, isText = false;
+                        Header encoding = method.getResponseHeader("Content-Encoding");
+                        if (encoding != null && encoding.getValue().contains("gzip")) {
+                            decompress = true;
+                        }
```

Comment:
You should not have to be concern with handling of encoding.  This knowledge should be in HttpServices.  
Could you create a common HttpServices.getInputStream() so that this and OutputStreamHandler can use?